server/utils/uploadImage.js

```diff
@@ -0,0 +1,60 @@
+import multer from "multer";
+import multerS3 from "multer-s3";
+import { S3Client } from "@aws-sdk/client-s3";
+import path from "node:path";
+
+// create s3 instance using S3Client
+// (this is how we create s3 instance in v3)
+const s3 = new S3Client({
+	credentials: {
+		accessKeyId: process.env.ACCESS_KEY_ID, // store it in .env file to keep it safe
+		secretAccessKey: process.env.SECRET_ACCESS_KEY,
+	},
+	region: "eu-west-1", // this is the region that you select in AWS account
+});
+
+const s3Storage = multerS3({
+	s3: s3, // s3 instance
+	bucket: "cyf-images-fp", // change it as per your project requirement
+	acl: "public-read", // storage access type
+	metadata: (req, file, cb) => {
+		cb(null, { fieldname: file.fieldname });
+	},
+	key: (req, file, cb) => {
+		const fileName =
+			Date.now() + "_" + file.fieldname + "_" + file.originalname;
+		cb(null, fileName);
+	},
+});
+
+// function to sanitize files and send error for unsupported files
+function sanitizeFile(file, cb) {
+	// Define the allowed extension
+	const fileExts = [".png", ".jpg", ".jpeg", ".gif", ".svg"];
+
+	// Check allowed extensions
+	const isAllowedExt = fileExts.includes(
+		path.extname(file.originalname.toLowerCase())
+	);
+
+	// Mime type must be an image
+	const isAllowedMimeType = file.mimetype.startsWith("image/");
+
+	if (isAllowedExt && isAllowedMimeType) {
+		return cb(null, true); // no errors
+	} else {
+		// pass error msg to callback, which can be displayed in frontend
+		cb("Error: File type not allowed!");
```

Comment:
We haven't discussed error handling but ideally error messages would be show to a user on the frontend so they understand what went wrong and can fix the problem and try again

Reply:
I've altered the message to be clearer in this commit:https://github.com/Innapoliakova/set/pull/50/commits/22ce2924421ae0cb032dc963d1a4d26a1c084de4

Reply:
Nice!